coreservices/partsrelationshipservice/prs/src/main/java/net/catenax/prs/requests/VinPartsTreeRequest.java

```diff
@@ -0,0 +1,42 @@
+//
+// Copyright (c) 2021 Copyright Holder (Catena-X Consortium)
+//
+// See the AUTHORS file(s) distributed with this work for additional
+// information regarding authorship.
+//
+// See the LICENSE file(s) distributed with this work for
+// additional information regarding license terms.
+//
+package net.catenax.prs.requests;
+
+import com.catenax.partsrelationshipservice.dtos.PartsTreeView;
+import io.swagger.v3.oas.annotations.Parameter;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Value;
+import net.catenax.prs.util.SampleDataHelper;
+
+import java.util.Optional;
+
+import static io.swagger.v3.oas.annotations.enums.ParameterIn.PATH;
+
+/**
+ * Parameter object for getPartsTreeByVin REST operation.
+ */
+@Value
+@Builder(toBuilder = true)
+@AllArgsConstructor
+@SuppressWarnings("PMD.CommentRequired")
+public class VinPartsTreeRequest {
+    @Parameter(description = "Vehicle Identification Number", example = SampleDataHelper.SERIAL_NUMBER_BMW_PART_1, in = PATH, required = true)
+    private final String vin;
```

Comment:
```suggestion
   String vin;
```

private final is redundant as we have @value here. Applies to others as well.

Reply:
Working on this

Reply:
final is redundant, but not private. Fixed :)